p11-kit/pkcs11-gnu.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2018 Red Hat, Inc
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above
+ *       copyright notice, this list of conditions and the
+ *       following disclaimer.
+ *     * Redistributions in binary form must reproduce the
+ *       above copyright notice, this list of conditions and
+ *       the following disclaimer in the documentation and/or
+ *       other materials provided with the distribution.
+ *     * The names of contributors to this software may not be
+ *       used to endorse or promote products derived from this
+ *       software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * Author: Daiki Ueno <dueno@redhat.com>
+ */
+
+/*
+ * This is to check that the exposed API can be used through the PKCS
+ * #11 GNU calling convention.
+ */
+
+#include <stdlib.h>
+
+#define CRYPTOKI_GNU 1
+#define P11_KIT_FUTURE_UNSTABLE_API 1
+
+#include <p11-kit/p11-kit.h>
+#include <p11-kit/iter.h>
+#include <p11-kit/pin.h>
+#include <p11-kit/uri.h>
```

Comment:
I'm wondering whether the compiler could ignore some type errors if the code is not used. nevertheless this seems to catch the existing issue.

Reply:
I added references to the defined symbols so that the declarations are actually used.
Also this file is split into 3 (generated): iter.gnu.c, pin.gnu.c, and uri.gnu.c, so that the type declarations in other files do not take effect.